src/js/yeldaWebchatInjector.js

```diff
@@ -775,40 +775,109 @@ class YeldaChat {
+      return  `${urlBase}/settings/${data.settingId}/chatBubble?${envQuery}`
     }
 
-    return  `${data.assistantUrl}/assistants/${data.assistantId}/chatBubble/${data.locale}`
+    return  `${urlBase}/${data.assistantId}/chatBubble/${data.locale}?${envQuery}`
   }
 
   /**
    * Update assistantImage with assistant settings from backend if any
    * @param {Object} data { assistantUrl, assistantId, settingId }
    * @return {Promise}
    */
-  getAssistantSettings(data) {
+  getAssistantSettings(data, { isFallback = false, customTimeout = null } = {}) {
+    // Set `this` to be able to do some recursion in onreadystatechange
+    const self = this;
     return new Promise((resolve, reject) => {
       try {
         const xhr = new XMLHttpRequest()
-        const url = this.getAssistantSettingsUrl(data)
 
+        // Get the url base depending on if we are doing the basic request (external chatbubble endpoint) 
+        // or the fallback one (chatbubble endpoint in Yelda)
+        const urlBase = this.getChatBubbleUrlBase(data, isFallback)
+        const url = this.getAssistantSettingsUrl(data, urlBase)
+        
         xhr.open('GET', url)
+        if (customTimeout) {
+          xhr.timeout = customTimeout
+        }
         xhr.send()
 
         // Bind and call are necessary to pass the "this" to the callback function
         xhr.onreadystatechange = function() {
           if (xhr.readyState === 4) {
-            const webchatSettings = xhr.responseText ? JSON.parse(xhr.responseText).data : null
+            const responseText = xhr.responseText ? JSON.parse(xhr.responseText) : null
+            // @TODO check if we have still have case where the settings are set in a data field 
+            // like { data: { ...webchatSettings} } instead of { ...webchatSettings }
+            // From last tests, data is received as { ...webchatSettings } from both the external endpoint and the yelda endpoint
+            const webchatSettings = responseText && (responseText.data || responseText)
+
+            /*
+              1. If !fallback === true, it means that this is the initial request to the external chatbubble endpoint
+                 If it does not return any webchatSettings, it might be because they were not set in Redis yet
+                 So let's call the yelda endpoint that will find the setting (if it exists) and set it in Redis
+              2. If !fallback === false, it means that this is the fallback request
+                 So if we still dont have any webchatSettings, it means that the does not have any webchat settings, 
+                 which should not happen except in case of specific error since the webchat settings are created on assistant creation
+            */
+            if (!webchatSettings && !isFallback) {
+              // Set isFallback to true to be sure to not end up in an infinite loop
+              resolve(self.getAssistantSettings(data, { isFallback: true }))
+            }
```

Comment:
:gear: too complex. Instead of multiple negative check, reverse the logic : positive or positive
:gear: / :warning: good practice : add a return before the resolve()

```js
            // We're good :tada: 
            if (webchatSettings) {
              return resolve(webchatSettings)
            }
            
            /*
             * We did not received the expected webchatSettings
             * If we were already in fallback mode, it means there is no webchat settings for the given init information
             */
            if (isFallback)
              return resolve(null)
            } 
            
            /* 
             * If the initial request to the external chatbubble endpoint did not returned data on time
             * it might be because they were not set in Redis yet
             * So let's call the yelda endpoint (fallback endpoint) 
             * that will find the setting (if it exists) and set it in Redis
             */
            return resolve(self.getAssistantSettings(data, { isFallback: true }))
```